.github/workflows/publish.yml

```diff
@@ -0,0 +1,63 @@
+name: Publish and deploy
+
+# Note: Adapted from terra-data-catalog.
+# For future functionality like 'publish to Artifactory', 'push to GCR', ..., see the data catalog version of this file.
+
+on:
+  create:
+  workflow_dispatch:
+    inputs:
+      tagOverride:
+        description: 'Specify the tag to use for this publish'
+        required: true
+
+env:
+  # NB: Unlike terra-data-catalog, this doesn't match the repo name
+  SERVICE_NAME: composite-batch-analysis-service
+  DOCKERHUB_ORG: broadinstitute
+
+jobs:
+  publish-job:
+    if: startsWith(github.ref, 'refs/tags/') || github.event.inputs.tagOverride != ''
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'gradle'
+
+      - name: Parse tag
+        # Sets up a step output called TAG with the appropriate tag
+        id: tag
+        run: |
+          TAG_OVERRIDE=${{ github.event.inputs.tagOverride }}
+          if [[ ! -z "${TAG_OVERRIDE}" ]]
+          then
+            echo ::set-output name=TAG::${TAG_OVERRIDE}
+          else
+            echo ::set-output name=TAG::$(git describe --tags)
+          fi
+
+      - name: Construct docker image name and tag
+        id: image-name
+        run: |
+          echo ::set-output name=DOCKERHUB_NAME::${DOCKERHUB_ORG}/${SERVICE_NAME}:${{ steps.tag.outputs.TAG }}
+
+      - name: Build image locally with jib
+        run: |
+          ./gradlew --build-cache :service:jibDockerBuild \
+          --image=${{ steps.image-name.outputs.DOCKERHUB_NAME }} \
+          -Djib.console=plain
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: dsdejenkins
+          password: ${{ secrets.DSDEJENKINS_DOCKERHUB_PASSWORD }}
+
+      - name: Push dockerhub image
+        run: docker push ${{ steps.image-name.outputs.DOCKERHUB_NAME }}
```

Comment:
I had recently learnt ([Slack conversation](https://broadinstitute.slack.com/archives/C0DSD41QT/p1651166000447249)) that GCR is the standard place to host images. Should we also consider pushing CBAS image to GCR?

Reply:
Yes I think that's certainly going to be the long term plan. For now (and since we'll also want to use this in Azure), docker seemed like a good first step.